typespec-extension/src/code-model-builder.ts

```diff
@@ -681,34 +672,22 @@ export class CodeModelBuilder {
    * @returns filtered api-versions
    */
   private getFilteredApiVersions(
-    pinnedApiVersion: Version | undefined,
-    versions: Version[],
+    pinnedApiVersion: string | undefined,
+    versions: string[],
     excludePreview: boolean = false,
-  ): Version[] {
+  ): string[] {
     if (!pinnedApiVersion) {
       return versions;
     }
     return versions
       .slice(0, versions.indexOf(pinnedApiVersion) + 1)
-      .filter((version) => !excludePreview || !isStable(pinnedApiVersion) || isStable(version));
+      .filter((version) => !excludePreview || pinnedApiVersion.includes("preview") || !version.includes("preview"));
   }
 
-  /**
-   * `@armProviderNamespace` currently will add a default server if not defined globally:
-   * https://github.com/Azure/typespec-azure/blob/8b8d7c05f168d9305a09691c4fedcb88f4a57652/packages/typespec-azure-resource-manager/src/namespace.ts#L121-L128
-   * TODO: if the synthesized server has the right hostParameter, we can use that instead
-   *
-   * @param server returned by getServers
-   * @returns whether it's synthesized by `@armProviderNamespace`
-   */
-  private isArmSynthesizedServer(server: HttpServer): boolean {
-    return this.isArm() && (!server.parameters || server.parameters.size == 0);
-  }
-
-  private needToSkipProcessingOperation(operation: Operation, clientContext: ClientContext): boolean {
+  private needToSkipProcessingOperation(operation: Operation | undefined, clientContext: ClientContext): boolean {
     // don't generate protocol and convenience method for overloaded operations
     // issue link: https://github.com/Azure/autorest.java/issues/1958#issuecomment-1562558219 we will support generate overload methods for non-union type in future (TODO issue: https://github.com/Azure/autorest.java/issues/2160)
-    if (getOverloadedOperation(this.program, operation)) {
+    if (operation && getOverloadedOperation(this.program, operation)) {
       this.trace(`Operation '${operation.name}' is temporary skipped, as it is an overloaded operation`);
       return true;
     }
```

Comment:
Given the func, it seems to say "do not skip this operation, if it is undefined"?

Reply:
right, better to add check to return true for `operation === undefined`, I will update code.